src/main.py

```diff
@@ -909,14 +915,14 @@ def _verify_timeline(self, db_state, zk_state):
         # If it does, but there is no info on replicas,
         # close local PG instance.
         if tli_res:
-            if zk_state['replics_info_written'] is False:
+            if not just_check and zk_state['replics_info_written'] is False:
                 logging.error('Some error with ZK.')
                 # Actually we should never get here but checking it just in case.
                 # Here we should end iteration and check and probably close primary
                 # at the begin of primary_iter
                 return None
         # If ZK does not have timeline info, write it.
-        elif zk_state[self.zk.TIMELINE_INFO_PATH] is None:
+        elif zk_state[self.zk.TIMELINE_INFO_PATH] is None and not just_check:
             logging.warning('Could not get timeline from ZK. Saving it.')
             self.zk.write(self.zk.TIMELINE_INFO_PATH, db_state['timeline'])
         # If there is a mismatch in timeline:
```

Comment:
You can end up with `tli_res is False` (https://github.com/yandex/pgconsul/pull/24/files#diff-2e5ad92c43aa96cc3a9cef6c6aec998b216f1379c43b1f651013d25e55989312R934) while just_check is True.
In that case you will be doing modifications in Zookeeper while.
That is wrong because you don't have a lock and information that noone has it can be stale.